dlt/transformations/lineage.py

```diff
@@ -0,0 +1,178 @@
+    DataType.Type.NVARCHAR: "text",
+    DataType.Type.TEXT: "text",
+    DataType.Type.VARCHAR: "text",
+    DataType.Type.NAME: "text",
+    # SIGNED_INTEGER
+    DataType.Type.BIGINT: "bigint",
+    DataType.Type.INT: "bigint",
+    DataType.Type.INT128: "bigint",
+    DataType.Type.INT256: "bigint",
+    DataType.Type.MEDIUMINT: "bigint",
+    DataType.Type.SMALLINT: "bigint",
+    DataType.Type.TINYINT: "bigint",
+    # UNSIGNED_INTEGER
+    DataType.Type.UBIGINT: "bigint",
+    DataType.Type.UINT: "bigint",
+    DataType.Type.UINT128: "bigint",
+    DataType.Type.UINT256: "bigint",
+    DataType.Type.UMEDIUMINT: "bigint",
+    DataType.Type.USMALLINT: "bigint",
+    DataType.Type.UTINYINT: "bigint",
+    # other INTEGER
+    DataType.Type.BIT: "bigint",
+    # FLOAT
+    DataType.Type.DOUBLE: "double",
+    DataType.Type.FLOAT: "double",
+    # DECIMAL
+    DataType.Type.BIGDECIMAL: "decimal",
+    DataType.Type.DECIMAL: "decimal",
+    DataType.Type.DECIMAL32: "decimal",
+    DataType.Type.DECIMAL64: "decimal",
+    DataType.Type.DECIMAL128: "decimal",
+    DataType.Type.DECIMAL256: "decimal",
+    DataType.Type.MONEY: "decimal",
+    DataType.Type.SMALLMONEY: "decimal",
+    DataType.Type.UDECIMAL: "decimal",
+    DataType.Type.UDOUBLE: "decimal",
+    # TEMPORAL
+    DataType.Type.DATE: "date",
+    DataType.Type.DATE32: "date",
+    DataType.Type.DATETIME: "date",
+    DataType.Type.DATETIME2: "date",
+    DataType.Type.DATETIME64: "date",
+    DataType.Type.SMALLDATETIME: "date",
+    DataType.Type.TIMESTAMP: "timestamp",
+    DataType.Type.TIMESTAMPNTZ: "timestamp",
+    DataType.Type.TIMESTAMPLTZ: "timestamp",
+    DataType.Type.TIMESTAMPTZ: "timestamp",
+    DataType.Type.TIMESTAMP_MS: "timestamp",
+    DataType.Type.TIMESTAMP_NS: "timestamp",
+    DataType.Type.TIMESTAMP_S: "timestamp",
+    DataType.Type.TIME: "time",
+    DataType.Type.TIMETZ: "time",
+    # binary
+    DataType.Type.VARBINARY: "binary",
+    # BOOLEAN
+    DataType.Type.BOOLEAN: "bool",
+    # UKNOWN
+    DataType.Type.UNKNOWN: None,
+}
+
+
+def to_sqlglot_type(
+    column: TColumnSchema, table: TTableSchema, type_mapper: DataTypeMapper, dialect: str
+) -> DATA_TYPE:
+    if not column.get("data_type"):
+        return None
+
+    destination_type = type_mapper.to_destination_type(column, cast(PreparedTableSchema, table))
+
+    # TODO modify nullable arg_types on destination_type
+    sqlglot_type = DataType.build(destination_type, dialect=dialect)
+    # TODO verify how nullable is used exactly in sqlglot
+    if column.get("nullable") is not None:
+        sqlglot_type.arg_types["nullable"] = column["nullable"]
+
+    # TODO refine what metadata is tied to the DataType vs the Column expression
+    sqlglot_type._meta = {k: v for k, v in column.items() if k not in ["data_type", "name"]}
+    return sqlglot_type
+
+
+def from_sqlglot_type(column: sge.Column) -> TColumnSchema:
+    col = {
+        "name": column.output_name,
+        **column.type.meta,
```

Comment:
we need some minimal rules on type propagation. IMO the minimum are:
In general case
- we carry nullability, precision and scale
- also unique, hard_delete and incremental
we drop anything else, including primary keys etc.

In special case, where we just copy table (all columns included, only where clause allowed, no aggregates on columns):
- we copy all hints
We can simplify this by just allowing a SELECT - WHERE LIMIT